optional-option-result.md

```diff
@@ -0,0 +1,126 @@
+- Topic Name: Optional Option/Result
+- Start Date: 2022-06-25
+- RFC PR: [vlang/rfcs#00000](https://github.com/vlang/rfcs/pull/00000)
+- V Issue: [vlang/v#00000](https://github.com/vlang/v/issues/00000)
+
+# Summary
+
+The language currently does not provide a way to let developers handle memory allocation issues.
+Many developers are happy with the state of things: memory allocation failures are rare and a panic is acceptable.
+However the language could (and probably should) provide a way to handle these issues.
+
+A way to achieve this goal would be to return option/result without requiring the caller to handle failure scenarios.
```

Comment:
*V* already has a universal mechanism for that - namely not writing `panic( ... )` but `return error( ... )` instead. At the caller site you write only `my_mem_allocating_fn()!` (notice the `!` at the end - as of now it is [not clear whether `!` will become `?` eventually](https://github.com/vlang/v/pull/14140#issuecomment-1115944547 )).

If there is any issue with suffix `!` (or `?`), then it is a bug.

But if there is not, then what is the motivation for this RFC?